x-pack/apm-server/sampling/config_test.go

```diff
@@ -14,73 +14,76 @@ import (
 	"github.com/elastic/apm-server/elasticsearch"
 	"github.com/elastic/apm-server/model"
 	"github.com/elastic/apm-server/x-pack/apm-server/sampling"
+	"github.com/elastic/apm-server/x-pack/apm-server/sampling/pubsub"
 )
 
 func TestNewProcessorConfigInvalid(t *testing.T) {
 	var config sampling.Config
-	assertInvalidConfigError := func(expectedError string) {
+	assertInvalidConfigError := func(expectedError error) {
 		t.Helper()
 		agg, err := sampling.NewProcessor(config)
-		require.Error(t, err)
 		require.Nil(t, agg)
-		assert.EqualError(t, err, "invalid tail-sampling config: "+expectedError)
+		require.ErrorIs(t, err, expectedError)
+		assert.ErrorContains(t, err, "invalid tail-sampling config: ")
```

Comment:
So I think this is actually a step backwards for the test. The intention of the test was to verify that the output is readable and useful in identifying the error. These assertions would still pass if you had a badly formatted error like: `BeatID unspecified invalid tail-sampling config: `.

Maybe that's a bit contrived, but my point is more that the test was fine as it was, and there's really no need for an `errors.Is` level assertion here. Could you please revert to `assert.EqualError`, and show what the complete error messages look like now?